libs/service/service.go

```diff
@@ -52,6 +40,18 @@ type Service interface {
 	Wait()
 }
 
+// ServiceImplementation describes the implementation that the
+// BaseService implementation wraps.
+type ServiceImplementation interface { // nolint:revive
```

Comment:
Can we name this more simply `Implementation`, given we're already in the `service` package? (`service.Implementation` vs. `service.ServiceImplementation`)